src/plots/cartesian/axes.js

```diff
@@ -129,12 +173,13 @@ axes.coerceRef = function(containerIn, containerOut, gd, attr, dflt, extraOption
  * - for other types: coerce them to numbers
  */
 axes.coercePosition = function(containerOut, gd, coerce, axRef, attr, dflt) {
-    var cleanPos, pos;
-
-    if(axRef === 'paper' || axRef === 'pixel') {
+    var cleanPos, pos, axRefType;
+    axRefType = axes.getRefType(axRef);
+    if(axRefType === 'paper' || axRefType === 'pixel' || axRefType === 'domain') {
         cleanPos = Lib.ensureNumber;
         pos = coerce(attr, dflt);
     } else {
+        // if axRef is 'range' or undefined we will end up here
```

Comment:
Won't we fail if we get here with `undefined`? We won't find an `ax` so `ax.fraction2r` will error. In which case the condition above could be simplified to `if(axRefType !== 'range')`.